language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/codegen/Update.java

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.javaapi.data.codegen;
+
+import com.daml.ledger.javaapi.data.Command;
+
+public abstract class Update<R> {
```

Comment:
I think that `Update` can be a `final class`, and not the root of a hierarchy.

The best way to figure that out would be to try to define a Java version of [this utility](https://github.com/digital-asset/daml/issues/14312#issuecomment-1266462855) in the `rxjava.CommandClient`, perhaps defined in terms of `submitAndWaitForTransaction` (just one of the overloads, to test), and see if you can handle every `Command` case and deliver a `Single<T>` or `Single<Something<T>>` (where `Something` is to be defined and is a functor over `T`).